### ISSUE TITLE
Datagrid columns cannot be resized with keyboard

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3372]
Our datagrid header columns can be resized only with the mouse, we need to support keyboard interactions for this.

Need to verify correct approach, but seems like a pattern would be to 'pick up' the handle with the space bar when it has focus, and right/left arrows will change the size incrementally.

**Comment by gnomeontherun** [comment]
@Shijir might be able to shed more light on this with regards to how DnD utilities can help solve this.

**Comment by clane** [comment]
I think the arrow keys should be able to resize a column as soon as the handle gains focus as seen at https://salesforce-ux.github.io/dnd-a11y-patterns/#/resize?_k=59mvpi. This example from SalesForce could be improved my making the handle visible at all times rather than being visible only when focused.

**Comment by clane** [comment]
Taking a closer look at the Salesforce example I found that it does not support arrow key navigation between cells as expected of an ARIA Grid. We are going to have to place visible handles within the cells and allow users to focus the handles with technique described at [Editing and Navigating Inside a Cell](https://www.w3.org/TR/wai-aria-practices-1.1/#gridNav_inside) for navigating inside a cell

**Comment by bashibozuk** [comment]
Guys, this PR https://github.com/vmware/clarity/pull/4368 seems to fix this issue. Are any other actions than approval and merge required ?

**Comment by hippee-lee** [comment]
That PR is under review. It looks like it is getting close but there are a few things the author needs to address before it can be merged in.

**Comment by Shijir** [comment]
Closed by https://github.com/vmware/clarity/pull/4535

**Comment by gnomeontherun** [comment]
Closing as resolved by #4535.
